src/eval/methods.rs

```diff
@@ -185,6 +186,38 @@ pub fn call(
                     }
                     _ => return missing(),
                 }
+            } else if let Some(&datetime) = dynamic.downcast::<Datetime>() {
+                match method {
+                    "display" => {
+                        let pattern = args.eat()?;
+                        match datetime.display(pattern) {
+                            Ok(d) => Value::Str(Str::from(d)),
+                            Err(msg) => bail!(args.span, msg),
+                        }
+                    }
+                    "year" => datetime
+                        .date()
```

Comment:
maybe we could have these as real methods on the datetime. In general, I try to have as much of the logic in an actual method and only do any necessary casting here. so these could return `Option<{integer type}>` and then `into` should just do the job for you. (you might have to add impls in cast.rs if the integer type you're using doesn't have them yet.)

Reply:
I initially didn't do this because of the boilerplate, but I changed it now.